From 81eddfd468230494aa482db4003be9957f0cc9de Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Thu, 25 Jun 2020 14:35:22 +0300
Subject: [PATCH] PONRTSYS-7672:
 omci-channel-stuck-in-ds-if-board-powered-up-under-user-traffic-WA

Workaround to exclude QoS queue 0 from the database

Signed-off-by: Emergi Idan <idan.emergi@intel.com>
---
 drivers/net/ppv4/pp-api/pp-qos-mgr/pp_qos_utils.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ppv4/pp-api/pp-qos-mgr/pp_qos_utils.c b/drivers/net/ppv4/pp-api/pp-qos-mgr/pp_qos_utils.c
index 9b91b5759857..ebce4a565946 100644
--- a/drivers/net/ppv4/pp-api/pp-qos-mgr/pp_qos_utils.c
+++ b/drivers/net/ppv4/pp-api/pp-qos-mgr/pp_qos_utils.c
@@ -2022,8 +2022,8 @@ s32 qos_pools_init(struct pp_qos_dev *qdev, u32 max_ports, u32 max_queues,
 	start_q_idx = 0;
 	max_q_idx = max_queues - 1 - reserved_queues - synq_queues;
 	qdev->rlms = free_rlm_init(max_q_idx,
-				   max_queues - reserved_queues - synq_queues);
-	qdev->rlm_ranges[RLM_RANGES_NORMAL].start = 0;
+				   max_queues - reserved_queues - synq_queues - 1);
+	qdev->rlm_ranges[RLM_RANGES_NORMAL].start = 1;
 	qdev->rlm_ranges[RLM_RANGES_NORMAL].end = max_q_idx;
 
 	if (!qdev->rlms)
@@ -4380,9 +4380,10 @@ s32 qos_dbg_qm_stat_show(struct pp_qos_dev *qdev, struct seq_file *s)
 
 		cur_stat = &qminfo->qstat[0];
 		for (j = 0; j < qminfo->num_queues; ++j) {
-			QOS_DBG_PRINT(s, "%u\t\t%u\t\t%u\t\t%u\t\t%u\n",
+			QOS_DBG_PRINT(s, "%u\t\t%u(%u)\t\t%u\t\t%u\t\t%u\n",
 				      cur_stat->qid,
 				      cur_stat->qstat.queue_size_entries,
+				      cur_stat->qstat.qmgr_num_queue_entries,
 				      cur_stat->qstat.total_accepts,
 				      cur_stat->qstat.total_drops,
 				      cur_stat->qstat.total_red_dropped);
